core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/MethodConfigResolver.scala

```diff
@@ -30,13 +30,23 @@ object MethodConfigResolver {
   }
 
   private def getArrayResult(inputName: String, seq: Iterable[AttributeValue]): SubmissionValidationValue = {
-    val notNull = seq.filter(v => v != null && v != AttributeNull)
-    val attr = if (notNull.isEmpty) Option(AttributeValueEmptyList) else Option(AttributeValueList(notNull.toSeq))
+    val notNull: Seq[AttributeValue] = seq.filter(v => v != null && v != AttributeNull).toSeq
+    val attr = notNull match {
+      case Nil => Option(AttributeValueEmptyList)
+      //GAWB-2509: don't pack single-elem RawJson array results into another layer of array
+      //NOTE: This works, except for the following situation: a participant with a RawJson double-array attribute, in a single-element participant set.
+      // Evaluating this.participants.raw_json on the pset will incorrectly hit this case and return a 2D array when it should return a 3D array.
+      // The true fix for this is to look into why the slick expression evaluator wraps deserialized AttributeValues in a Seq, and instead
+      // return the proper result type, removing the need to infer whether it's a scalar or array type from the WDL input.
+      case AttributeValueRawJson(JsArray(_)) +: Seq() => Option(notNull.head)
```

Comment:
Here's the gross hack: if our target WDL type is an array, and we've got some raw json that's an array, then just return that raw json.